src/test/nn/transformer/model_test.py

```diff
@@ -229,6 +245,62 @@ def test_tensor_parallel_transformer(backend: str, architecture: str, tmp_path):
     )
 
 
+def run_context_parallel_transformer(checkpoint_dir, outputs_path, architecture: str):
+    device = get_default_device()
+    config = get_transformer_config(architecture, dtype=torch.bfloat16)
+    config.block.attention.use_flash = True
+
+    mesh = init_device_mesh(
+        device.type,
+        (get_world_size(),),
+        mesh_dim_names=("cp",),
+    )
+
+    model = config.build()
+    model.apply_cp(mesh["cp"], RingAttentionLoadBalancerType.zig_zag)
+    model.init_weights(device=device, max_seq_len=512)
+    load_model_and_optim_state(checkpoint_dir, model)
+
+    input_ids = get_transformer_inputs().to(device)
+    local_logits = model(input_ids=input_ids)
+    logits = DTensor.from_local(local_logits, mesh, (Shard(1),))
+
+    og_logits = torch.load(outputs_path, map_location=device)
+    torch.testing.assert_close(og_logits, get_full_tensor(logits))
+
+
+@requires_multi_gpu
+@requires_flash_attn
+@pytest.mark.parametrize("architecture", ["olmo2"])
+@pytest.mark.skip("known precision issues with ring-flash-attn")
```

Comment:
I gave up on getting this test to pass for now. I'm pretty sure it's just precision issues with ring-flash-attn. Even if you remove RoPE it fails.